microsoft-azure-storage/src/com/microsoft/azure/storage/Constants.java

```diff
@@ -1207,8 +1207,8 @@
     /**
      * The maximum size, in bytes, of a given stream mark operation.
      */
-    // Note if BlobConstants.MAX_SINGLE_UPLOAD_BLOB_SIZE_IN_BYTES is updated then this needs to be as well.
-    public static final int MAX_MARK_LENGTH = 256 * MB;
+    //This actually acquire the buffer into memory until file gets copied
+    public static final int MAX_MARK_LENGTH = 8 * MB;
```

Comment:
Why did you change this and MAX_SINGLE_UPLOAD_SIZE? Can you leave these values and just modify the blob request options?

Reply:
Thank you,

Surely will do that and I think that will work too, however the default value/implementation causes issue for bug file.